Modules/_decimal/_decimal.c

```diff
@@ -3423,16 +3425,11 @@ dec_format(PyObject *dec, PyObject *args)
     PyObject *fmtarg;
     PyObject *context;
     mpd_spec_t spec;
-    char const *fmt;
-    char *fmt_copy = NULL;
+    char *fmt;
```

Comment:
It is not related to handling the 'z' format. It is safer to keep `fmt` a const string.

Reply:
I disagree for two reasons:

The revert of the z-format change should be clean. Then, `fmt` is allocated iff `replace_fillchar` is true, so I do not see any safety issues in general. `replace_fillchar` is more informative than `fmt_copy` (and the 3.9 code has been tested with my own test suite patches that inject all allocation failures).

Reply:
You should be care to not modify the content referred by `fmt` without a help from the C compiler, but otherwise it is not a big deal.

Reply:
While I don't doubt the code prior to the original `z` format PR was correct, I added this const-correctness because the PR was doing yet more hacking of the format string, and I wanted the overall code to be clear and supported by compile-time checks as much as possible.

I'm OK to revert this change, but I think there are advantages to the new implementation.